src/features/notifications/components/NotificationLink.tsx

```diff
@@ -0,0 +1,16 @@
+import { FC, PropsWithChildren } from 'react';
+import { Link } from 'react-router-dom';
+
+interface Props {
+  path: string;
+  notificationId: number;
+  handleMarkRead?: (notificationId: number) => void;
+}
+
+export const NotificationLink: FC<PropsWithChildren<Props>> = ({ children, path, notificationId, handleMarkRead }) => {
+  return (
+    <Link to={path} onClick={handleMarkRead ? () => handleMarkRead(notificationId) : undefined}>
+      {children}
+    </Link>
+  );
+};
```

Comment:
`NotificationLink` does not feel like it's doing enough to justify it being a separate component, lets just use a `Link`.